### PR TITLE
Routing API review changes

### PR DESCRIPTION
**Pull request by JamesNK**

### CODE REVIEW

**Comment by JamesNK** [comment]
I'm going to talk with y'all about feature branches tomorrow and How To Use Them, because this has lots of breaks for MVC.

**Comment by rynowak** [comment]
@JamesNK this should help https://github.com/aspnet/Universe/blob/master/docs/CrossRepoBreakingChanges.md

Just remember that you have to do it again for `master`.

File: benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/AsciiKeyedJumpTable.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 using System;
 using System.Collections.Generic;
 
-namespace Microsoft.AspNetCore.Routing.Matchers
```
Comment (rynowak) [inline benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/AsciiKeyedJumpTable.cs]: A whole bunch of these benchmarks files should be moving to a different folder also

File: src/Microsoft.AspNetCore.Routing/Matchers/AmbiguousMatchException.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 using System;
 using System.Runtime.Serialization;
 
-namespace Microsoft.AspNetCore.Routing.Matchers
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Routing/Matchers/AmbiguousMatchException.cs]: All of these files should be moving on disk as well

File: src/Microsoft.AspNetCore.Routing/RouteValuesBasedEndpointFinder.cs [hunk 3]
```diff
@@ -36,7 +35,7 @@ internal class RouteValuesBasedEndpointFinder : IEndpointFinder<RouteValuesBased
                 HandleChange);
         }
 
-        public IEnumerable<Endpoint> FindEndpoints(RouteValuesBasedEndpointFinderContext context)
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Routing/RouteValuesBasedEndpointFinder.cs]: context -> address

File: test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs [hunk 4]
```diff
@@ -1553,11 +1552,11 @@ public void TryGetLink_DiscardsAmbientValues_ForGenericKeys()
             Assert.Null(link);
         }
 
-        private RouteValuesBasedEndpointFinderContext CreateRouteValuesContext(
+        private RouteValuesAddress CreateRouteValuesContext(
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs]: context -> address

File: test/Microsoft.AspNetCore.Routing.Tests/Matchers/BarebonesMatcher.cs [hunk 5]
```diff
@@ -5,7 +5,7 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
 
-namespace Microsoft.AspNetCore.Routing.Matchers
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.Tests/Matchers/BarebonesMatcher.cs]: These files should also be moving 👍

**Pull request closed without merging**